File: Recursion/Code/NthStair.py
#Program to find the number of ways to reach the nth stair

def countWays(n):
    if (n==1 or n==2): #Base Case
        return n
    else:
        return countWays(n-1) + countWays(n-2) # using countWays(n-1) and countWays(n-2) are "RECURSIVE CALLS"
        # adding the two together is the "SMALL CALCULATION"

N = int(input("Enter a positive number : "))
print("Number of ways to reach "+str(N)+"th Stair are = ",countWays(N))

File: Recursion/Code/NthStair.c
//Program to find the number of ways to reach the nth stair
#include <stdio.h>

int countWays(int n){
    if (n==1 || n==2){
        return n;
    }
    else{
        return countWays(n-1)+countWays(n-2);
    }
}

int main(){
    int N;
    printf("Enter a positive number : ");
    scanf("%d",&N);
    printf("Number of ways to reach %d",N);
    printf("th Stair are = %d\n",countWays(N));
}